### PR TITLE
Move from ioutil to os/io packages

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
With go1.16 we now default to the os/io packages instead of ioutil. This
should be reflected in the code, too.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: packages/deb/build.go [hunk 1]
```diff
@@ -131,7 +132,7 @@ func (c cfg) run() error {
 	var w []work
 
 	srcdir := filepath.Join(c.DistroName, c.Package)
-	dstdir, err := ioutil.TempDir(os.TempDir(), "debs")
+	dstdir, err := ioutil.TempDir("", "debs")
```
Comment (hasheddan) [inline packages/deb/build.go:135]: @saschagrunert should we use https://golang.org/pkg/os/#MkdirTemp?

File: pkg/binary/binary_test.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func GetTestHeaders() []TestHeader {
 
 // writeTestBinary Writes a test binary and returns the path
 func writeTestBinary(t *testing.T, base64Data *string) *os.File {
-	f, err := ioutil.TempFile(os.TempDir(), "test-binary-")
+	f, err := ioutil.TempFile("", "test-binary-")
```
Comment (hasheddan) [inline pkg/binary/binary_test.go:97]: Should we use https://golang.org/pkg/os/#CreateTemp?

File: pkg/license/download.go [hunk 3]
```diff
@@ -72,7 +73,7 @@ func (do *DownloaderOptions) Validate() error {
 		}
 		// And no cache dir was specified
 		if do.CacheDir == "" {
-			dir, err := ioutil.TempDir(os.TempDir(), "license-cache-")
+			dir, err := ioutil.TempDir("", "license-cache-")
```
Comment (hasheddan) [inline pkg/license/download.go:76]: Should we use https://golang.org/pkg/os/#MkdirTemp?

**Review by hasheddan — COMMENTED** [review]
@saschagrunert I think there are a number of places where we can move to `MkdirTemp()` and `CreateTemp()`. I didn't point out all of them to cut down on comments :)

File: packages/deb/build.go (hunk 1 above)
Comment (saschagrunert) [reply packages/deb/build.go]: Yes, fixed as suggested :+1:

File: pkg/binary/binary_test.go (hunk 2 above)
Comment (saschagrunert) [reply pkg/binary/binary_test.go]: Yes, fixed as suggested :+1:

File: pkg/license/download.go (hunk 3 above)
Comment (saschagrunert) [reply pkg/license/download.go]: Yes, fixed as suggested :+1:

**Comment by saschagrunert** [comment]
Corresponding PR to release-utils: https://github.com/kubernetes-sigs/release-utils/pull/8

**Review by hasheddan — APPROVED** [review]
/lgtm

**Pull request merged**